contracts/Items/DEFYForge.sol

```diff
@@ -0,0 +1,241 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.9;
+
+import "./IDEFYItems.sol";
+import "@openzeppelin/contracts/security/Pausable.sol";
+import "@openzeppelin/contracts/access/AccessControl.sol";
+
+contract DEFYForge is Pausable, AccessControl {
+    bytes32 public constant FORGER_ROLE = keccak256("FORGER_ROLE");
+
+    struct PrintJob {
+        IERC1155 itemContract;
+        address operativeAddress;
+        uint256[] inputMaterialsId;
+        uint256[] inputMaterialsAmount;
+        uint256 blueprintId;
+        uint256 outputPartId;
+        bool completed;
+    }
+
+    bytes32[] private printJobIds;
+    mapping(bytes32 => PrintJob) private printJobs;
```

Comment:
Can remove uint256p[] (bytes32[]) forgeJobIds array, as it would just be sequential 0, 1, 2, 3 etc...

Reply:
on further inspection, the ERC721 contract needs it because it removes ids when tokens are burned. This contract won't burn or remove printjobs, so can remove array of forgeJobIds, and get length / no. of jobs via the counter